dogs-initial/src/main/java/com/lambdaschool/dogsinitial/handler/RestExceptionHandler.java

```diff
@@ -0,0 +1,68 @@
+
+
+package com.lambdaschool.dogsinitial.handler;
+
+        import com.lambdaschool.dogsinitial.exception.ResourceNotFoundException;
+        import com.lambdaschool.dogsinitial.model.ErrorDetail;
+        import org.springframework.beans.TypeMismatchException;
+        import org.springframework.beans.factory.annotation.Autowired;
+        import org.springframework.context.MessageSource;
+        import org.springframework.http.HttpHeaders;
+        import org.springframework.http.HttpStatus;
+        import org.springframework.http.ResponseEntity;
+        import org.springframework.web.bind.annotation.ControllerAdvice;
+        import org.springframework.web.bind.annotation.ExceptionHandler;
+        import org.springframework.web.context.request.WebRequest;
+        import org.springframework.web.servlet.NoHandlerFoundException;
+        import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+        import javax.servlet.http.HttpServletRequest;
+        import java.util.Date;
+
+// bean shared across controller classes
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler
+{
+    @Autowired
+    private MessageSource messageSource;
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    public ResponseEntity<?> handleResourceNotFoundException(ResourceNotFoundException rnfe, HttpServletRequest request)
+    {
+        ErrorDetail errorDetail = new ErrorDetail();
+        errorDetail.setTimestamp(new Date().getTime());
+        errorDetail.setStatus(HttpStatus.NOT_FOUND.value());
+        errorDetail.setTitle("Resource Not Found");
+        errorDetail.setDetail(rnfe.getMessage());
+        errorDetail.setDeveloperMessage(rnfe.getClass().getName());
+
+        return new ResponseEntity<>(errorDetail, null, HttpStatus.NOT_FOUND);
+    }
+
+
+    @Override
+    protected ResponseEntity<Object> handleTypeMismatch(TypeMismatchException ex, HttpHeaders headers, HttpStatus status, WebRequest request)
+    {
+        ErrorDetail errorDetail = new ErrorDetail();
+        errorDetail.setTimestamp(new Date().getTime());
+        errorDetail.setStatus(HttpStatus.BAD_REQUEST.value());
+        errorDetail.setTitle(ex.getPropertyName());
+        errorDetail.setDetail(ex.getMessage());
+        errorDetail.setDeveloperMessage(request.getDescription(true));
+
+        return new ResponseEntity<>(errorDetail, null, HttpStatus.NOT_FOUND);
```

Comment:
Conventions for handling type mismatches vary, though to maintain consistency, you can use `HttpStatus.BAD_REQUEST` to return the HTTP status `400` error code as appropriate.